packages/builder/src/components/common/bindings/BindingSidePanel.svelte

```diff
@@ -295,10 +409,53 @@
           </div>
         {/if}
       {/if}
+
+      {#if selectedCategory === "Snippets" || search}
+        <div class="snippet-list">
+          {#if enableSnippets}
+            {#each filteredSnippets as snippet}
+              <li
+                class="snippet"
+                on:mouseenter={e => showSnippet(snippet, e.currentTarget)}
+                on:mouseleave={hidePopover}
+                on:click={() => addSnippet(snippet)}
+              >
+                {snippet.name}
+                <Icon
+                  name="Edit"
+                  hoverable
+                  newStyles
+                  size="S"
+                  on:click={e => editSnippet(e, snippet)}
+                />
+              </li>
+            {/each}
+          {:else if !search}
+            <div class="upgrade">
+              <Body size="S">
+                Snippets let you create reusable JS functions and values that
+                can all be managed in one place
+              </Body>
+              {#if enableSnippets}
+                <Button cta on:click={createSnippet}>Create snippet</Button>
+              {:else}
+                <UpgradeButton />
+              {/if}
+            </div>
+          {/if}
+        </div>
+      {/if}
     {/if}
   </Layout>
 </div>
 
+{#if showSnippetDrawer}
```

Comment:
This is done as it was the easier way to clean the state. Otherwise, when creating multiple snippets, the state would be persisted instead of starting from a clean slate everytime

Reply:
Was there any way to work around this? It's not a big problem, but unmounting the component like this when hiding means we lose the drawer close animation - it just instantly disappears with a small flash rather than fading out with an animation.

Not super important but would be a nice to have I guess. Happy to keep it this way since I assume you already tried resetting state instead, but just wanted to note the downside of doing it like this.